DSLink/Respond/DSLinkResponder.cs

```diff
@@ -0,0 +1,183 @@
+using System;
+using System.Threading.Tasks;
+using DSLink.Container;
+using DSLink.Nodes;
+using DSLink.Request;
+using Newtonsoft.Json.Linq;
+
+namespace DSLink.Respond
+{
+    public class DSLinkResponder : Responder
+    {
+        public DSLinkResponder(AbstractContainer link) : base()
+        {
+            Link = link;
+            NodeSerializer = new NodeSerializer(this);
+            SuperRoot = new Node("", null, Link);
+            SubscriptionManager = new SubscriptionManager(Link);
+            StreamManager = new StreamManager(Link);
+        }
+
+        public override void AddNodeClass(string name, Action<Node> factory)
+        {
+            lock (NodeClasses)
+            {
+                NodeClasses[name] = factory;
+            }
+        }
+
+        public override async Task<JArray> ProcessRequests(JArray requests)
+        {
+            var responses = new JArray();
+            foreach (var jToken in requests)
+            {
+                var request = (JObject) jToken;
+                switch (request["method"].Value<string>())
+                {
+                    case "list":
+                        ListMethod(responses, request);
```

Comment:
I feel like this could be much cleaner with polymorphism.

You could probably keep the switch statement, delegate to a factory, that would create the "Methods". It'd also allow you to split all those Methods individually, and unit test them maybe? Just a thought.